packages/form-js-carbon-styles/src/carbon-styles.js

```diff
@@ -0,0 +1,993 @@
+          background-color: var(--cds-field, var(--cds-field-01));
+          display: flex;
+          padding-right: var(--cds-spacing-05);
+        }
+
+        .fjs-input-group .fjs-input-adornment svg {
+          color: var(--cds-icon-primary);
+          cursor: pointer;
+        }
+
+        .flatpickr-wrapper {
+          height: 100%;
+          position: initial;
+        }
+
+        .fjs-timepicker.fjs-timepicker-anchor {
+          position: unset;
+        }
+
+        .flatpickr-calendar.static {
+          top: calc(100% + 3px);
+        }
+
+        .flatpickr-calendar .flatpickr-prev-month svg,
+        .flatpickr-calendar .flatpickr-next-month svg {
+          height: 16px;
+        }
+
+        .flatpickr-day.today {
+          position: relative;
+          color: var(--cds-link-primary);
+          font-weight: 600;
+          border-color: transparent;
+        }
+
+        .flatpickr-day.selected,
+        .flatpickr-day.today.selected,
+        .flatpickr-day.selected:hover,
+        .flatpickr-day.today.selected:hover {
+          background-color: var(--cds-button-primary);
+          color: var(--cds-text-on-color);
+        }
+
+        .flatpickr-day:focus {
+          outline: 2px solid var(--cds-focus);
+          outline-offset: -2px;
+        }
+
+        .flatpickr-day.selected:focus {
+          outline: 0.0625rem solid var(--cds-focus);
+          outline-offset: -0.1875rem;
+        }
+
+        .flatpickr-day:hover {
+          background: var(--cds-layer-hover, var(--cds-layer-hover-01));
+        }
+
+        .flatpickr-days,
+        .flatpickr-weekdays {
+          padding: unset;
+          width: unset;
+        }
+      }
+    }
+  `}
+`;
+
+const SELECT_STYLES = css`
+  ${({ theme }) => css`
+    .fjs-container {
+      .fjs-form-field-select .fjs-input-group {
+        ${getBaseInputStyles({ height: '2.5rem' })}
+        ${getSelectArrowStyles({
+    arrowRightPosition: 'var(--cds-spacing-05)',
+    color: theme.iconPrimary,
+  })}
+
+        .fjs-select-display {
+          display: flex;
+          align-items: center;
+        }
+
+        .fjs-select-arrow {
+          visibility: hidden;
+        }
+
+        .fjs-select-cross {
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          margin-right: var(--cds-spacing-04);
+          width: 2.5rem;
+          height: calc(2.5rem - 1px);
+
+          &:hover {
+            background-color: var(--cds-layer-hover, var(--cds-layer-hover-01));
+          }
+
+          svg {
+            color: var(--cds-icon-primary);
+          }
+        }
+
+        .fjs-input {
+          color: var(--cds-text-primary);
+          background-color: var(--cds-field, var(--cds-field-01));
+          border-radius: 0;
+          border: none;
+          border-bottom: 1px solid var(--cds-border-strong, var(--cds-border-strong-01));
+          height: 2.5rem;
+          ${theme.bodyShort01};
```

Comment:
These can also be replaced with CSS vars, just check for the CSS vars in the scope

Reply:
Do you mean we should remove all theme interpolations, so a ThemeProvider is not necessary anymore?

Reply:
Ah yeah, cf. to this comment: https://github.com/bpmn-io/form-js/pull/557#discussion_r1179607872

Let's do that, relaxes the integration a lot 👍

Reply:
By removing some `theme` occurrences, seems like I broke some styles: 
![image](https://user-images.githubusercontent.com/9433996/235079988-b914b39a-ee8e-4ea7-b4f3-ad48cf105553.png)

Maybe let's wait for the further review until I'm done with that.

Reply:
Ok I fixed at least this issue. Feel free to continue your visual review 👍 I will aim to remove all the theme accesses, without breaking further things.

Reply:
We just discussed that we can't remove the theme entirely as we need it for the background images (selects, numbers) - we can't use CSS variables over there.